security/secrets.md

```diff
@@ -0,0 +1,59 @@
+# Secrets Management
+
+Secrets and credentials (e.g. API keys, database logins, or other credentials) SHOULD be stored so they are not lost and can be shared appropriately. 
+
+If secrets are to be stored or shared, they MUST be stored or shared using [AWS Parameter Store](http://docs.aws.amazon.com/systems-manager/latest/userguide/systems-manager-paramstore.html).
```

Comment:
Not sure how I feel about they `MUST` be stored with the AWS Parameter Store. I see it as a `RECOMMENDED` because the secret/credential management principle is -- `MUST` comply with being stored securely which is the foundation of the guideline. Teams can/should decide what tool/workflow to use... not only relying on an AWS product. Similarly to Logging, it is a different topic but `Logging` with Loggly has some benefits over AWS and we should weigh pros/cons before committing to only AWS.

Seems like we are tying ourselves down with AWS for everything and `we` the entire team should have an agreement as well as @jvarghese01 

Thoughts @gkallenberg @thisisstephenbetts @nodanaonlyzuul @nonword @EdwinGuzman @jobinthomasnypl @holingpoon @ktp242 @emu47

Reply:
I appreciate your point. I think there's always a balance between team autonomy and standardization. I guess I'm leaning towards creating standards in a collaborative way (i.e. this process) so that we can benefit from having some common tooling, processes, etc. For example, imagine if every team decided to use a different secrets management product/process?

With that said, these standards are living, breathing documents so if a team _did_ find a better solution/tool they could then open a PR related to the standard and a healthy dialogue could ensue. The important thing is that teams/engineers feel they have a voice and place to propose improvements.

Does that make sense?

Reply:
In general, I like the words `SHOULD` rather than `RECOMMENDED`, `MUST` rather than `REQUIRED`, and `MAY` rather than `OPTIONAL`. The other documentation pages in this repo I've seen have been using this convention.  If there is a change in this convention of wording, *we* should decide as a *team*.

While I agree that we should not put all our eggs in one basket in AWS, I think the more complete picture would sound more like the following:

> If secrets are to be stored or shared, they MUST be stored or shared in one place, to signify one source of truth and to make resources easier to find, and we SHOULD be using AWS Parameter Store until another more sensible and usable technology is available.

We need to balance out having the freedom to decide what we prefer to use, versus having our resources scattered and hence hard to manage.

Reply:
I agree that the MUST/SHOULD would be better if rearranged in the secret's policy as per Ho-Ling's suggestion.

Reply:
@kfriedman -- Thanks for the feedback, it does make sense. I do agree with @holingpoon on the language change. As a breathing document, the proposal @holingpoon offered makes sense to me as the language/wording she used is clear about the usage of the `SHOULD` term as well as noting the following:

 `until another more sensible and usable technology is available`. 

@holingpoon I agree on using the term `SHOULD` over `RECOMMENDED`, there is no intention on my part to change it.

Reply:
Thanks for the great feedback @holingpoo, @rhernand3z, @gkallenberg. 

I made wording updates to reflect these comments. If it's okay with everyone, I'm going to omit the "until another more sensible and usable technology is available" wording because that's _always_ the case with any of these standards (i.e. if someone has a more "sensible" or "usable" suggestion, they should feel free to propose them in a PR).

If that's okay, let's proceed with merging this PR.